### ISSUE TITLE
NeoLauncher crashing and resetting home screen to default

### ISSUE THREAD

**Issue by Kurty00** [NeoApplications/Neo-Launcher#337]
## Description
Neo Launcher crashed and removed every widget, app shortcut, and page from the home screen.

## Expected Behavior
Neo Launcher should keep the home screen and ideally would not crash.

## Actual Behavior
Neo Launcher crashed and reset the desktop to default.

## Steps to Reproduce
While this has happened twice already, I unfortunately have no Idea how to reproduce the issue.

## Device and OS
Device: Fairphone 4
Android version: Android 12
Launcher version: 0.9.3
ROM: Stock Fairphone 4 ROM

## Logcat
Since the logcat was too long for GitHub, I've uploaded it here:
https://share.k00.eu/logcat_20230317_FP4.txt

**Comment by saulhdev** [comment]
It is related to #185. It will be closed when publish the next release.

**Comment by Nthoney** [comment]
I thought that the crashes were random, but now I found out they happen when my phone makes a Seedvault Backup.
Wish this helps.
Thanks for all Ur hard work!

**Comment by spart0n654** [comment]
I have this issue on pixel fold android 14 beta. I'm stock android without root and I can't reproduce it. It just happens randomly, 5-6 times per week. Thank goodness for backups for icon layouts but I have to still redo all my widgets.

**Comment by benplumley** [comment]
I just had this happen (it was triggered when I deleted a shortcut from my home screen) and it reoccurs immediately after applying my last backup, making the launcher unusable.

**Comment by tracewave** [comment]
This is also happening to me on Pixel 8 Pro. Cannot reproduce it, just happens and I don't know why.
After restoring a backup the widgets are not restored so I have to reconfigure all of them.

**Comment by CarlosEkisde** [comment]
This happened to me too when locking the desktop. but also when unlocking, and as @ltguillaume I've also seen the rows and columns count swap when the reset happens. 

However, I've managed to workaround the bug by adding and then deleting a row or a column just before locking or unlocking the home screen. Don't just delete without adding before because the launcher will just remove the icons from the row or colum it deleted.

Since the issue can be prevented by doing this (Has this happened to anyone using the same number of rows and columns? I'm personally using 10 columns and 13 rows)  I'd bet the launcher precisely is internally swapping the rows and columns count which would cause corruption in the database leading to the home screen to reset when locking the desktop.

**Comment by tracewave** [comment]
> Since the issue can be prevented by doing this (Has this happened to anyone using the same number of rows and columns? I'm personally using 10 columns and 13 rows) I'd bet the launcher precisely is internally swapping the rows and columns count which would cause corruption in the database leading to the home screen to reset when locking the desktop.

@CarlosEkisde Are you talking about Desktop/Grid size? I have it set to 5x5 and this happens to me at random. I don't even catch the crash. Sometimes I just take the phone and my icons and widgets are gone.

**Comment by CarlosEkisde** [comment]
> This seemed to work for me:
> 
>     1. Added a row
> 
>     2. Went back to homescreen to check if the row was added
> 
>     3. Removed the row and locked the desktop immediately after
> 
> 
> Thanks @CarlosEkisde!

You're welcome. I'm glad it worked. Remember this bug can trigger too when unlocking. The same workaround seems also to work also here. I keep a backup with NeoBackup app just in case anyway.

**Comment by MrMusic25** [comment]
Hello,

I am also experiencing this issue. Occasionally the launcher will reset to defaults, so I restore from most recent backup, but all my widgets are missing. Usually this is not a problem since it only happens maybe once a month or so randomly, but I have found a way to trigger it on my phone fairly consistently which has increased the annoyance of having to redo my desktop.

1. Open Google Maps
2. Start navigation to a destination
3. Press the home button (return to Neo Launcher)
4. Google Maps will start in picture-in-picture mode
5. Once Neo launcher finishes loading, the desktop layout has been reset

This is on a Sony Xperia 1 V running Android 14 (patched to March 1st, 2024 but this has been happening for months now). I am running Neo Launcher 0.9.3 (build 934). I believe it has to do specifically with the picture-in-picture mode. I tried it with the Twitch app and YouTube and their picture-in-picture did not cause a reset,, but Google Maps caused the reset 4/5 times in testing. Maybe it has to do with the weird form factor of the spawned window, but the picture-in-picture feature seems to trigger it a lot. I tried to disable it, but it is on a per-app basis so that would take too long.

I am going to try the desktop locking method mentioned above in hopes that it also solves my issue, once I finally get all my widgets back in place. I am a bit wary to use anything that uses picture-in-picture for now but hopefully locking solves it for me as well.
